ApplicationCode/Commands/RicExportContourMapToTextFeature.cpp

```diff
@@ -0,0 +1,287 @@
+                                                                const QString&                 undefinedValueLabel,
+                                                                bool                           excludeUndefinedValues )
+{
+    RifTextDataTableFormatter formatter( stream );
+    formatter.setTableRowLineAppendText( "" );
+    formatter.setTableRowPrependText( "" );
+    formatter.setCommentPrefix( "#" );
+    formatter.setHeaderPrefix( "" );
+    formatter.setDefaultMarker( undefinedValueLabel );
+
+    std::vector<RifTextDataTableColumn> header = {
+        RifTextDataTableColumn( "x" ),
+        RifTextDataTableColumn( "y" ),
+        RifTextDataTableColumn( "value" ),
+    };
+
+    formatter.header( header );
+
+    cvf::Vec2ui         numVerticesIJ    = contourMapProjection->numberOfVerticesIJ();
+    std::vector<double> xVertexPositions = contourMapProjection->xVertexPositions();
+    std::vector<double> yVertexPositions = contourMapProjection->yVertexPositions();
+
+    // Undefined values are positive inf in contour map projection.
+    double undefined = std::numeric_limits<double>::infinity();
+    for ( unsigned int j = 0; j < numVerticesIJ.y(); j++ )
+    {
+        for ( unsigned int i = 0; i < numVerticesIJ.x(); i++ )
+        {
+            double value = contourMapProjection->valueAtVertex( i, j );
+            if ( !( std::isinf( value ) && excludeUndefinedValues ) )
+            {
+                double x = xVertexPositions.at( i );
+                double y = yVertexPositions.at( j );
+                if ( !exportLocalCoordinates )
+                {
+                    x += contourMapProjection->origin3d().x();
+                    y += contourMapProjection->origin3d().y();
+                }
+
+                formatter.add( x );
+                formatter.add( y );
+                formatter.addValueOrDefaultMarker( value, undefined );
+                formatter.rowCompleted();
+            }
+        }
+    }
+
+    formatter.tableCompleted();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RicExportContourMapToTextFeature::setupActionLook( QAction* actionToSetup )
+{
+    actionToSetup->setText( "Export Contour Map to Text" );
+}
+
+RicfCommandResponse RicExportContourMapToTextFeature::execute()
+{
+    RicfCommandResponse response;
+    QStringList         errorMessages, warningMessages;
+
+    RiaApplication* app = RiaApplication::instance();
+
+    RimProject* proj = app->project();
+    CAF_ASSERT( proj );
+
+    std::vector<Rim3dView*> allViews;
+    proj->allViews( allViews );
+
+    Rim3dView* myView = nullptr;
+    for ( auto view : allViews )
+    {
+        if ( m_viewId == view->id() )
+        {
+            myView = view;
+        }
+    }
+
```

Comment:
Graceful return if no matching view is found

Reply:
Done.